### PR TITLE
use .NET libraries

### LINKED ISSUE #609: Idea - Malicious Libraries in TRJ_BANKER
Hi,

what is the best way to create rule/rules to detect these suspicious libraries?

2022-08-13 20_46_52-Configurações

Would it be a rule for each library?

Ref.: https://www.welivesecurity.com/2021/04/06/janeleiro-time-traveler-new-old-banking-trojan-brazil/

### PR DESCRIPTION
**Pull request by johnk3r**
Hello,

I'm creating this generic rule so we can discuss the best approach. I look forward to any recommendations/suggestions.

Closes #609

### CODE REVIEW

**Comment by mr-tz** [comment]
Great, thanks a lot. What do you think of breaking these up into multiple rules?
- use .NET library Costura.Fody
- use .NET library SharpVectors
- use .NET library Newtonsoft.Json
- use .NET library EncryptDecryptUtils
- use .NET library SharpClipboard

This allows us to identify 3rd party library usage more explicitly.

File: nursery/use-dotnet-library-newtonsoft.json.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+rule:
+  meta:
+    name: use .NET library Newtonsoft.Json
+    namespace: nursery
```
[the remark this answers is not in the document]
Comment (mr-tz) [reply nursery/use-dotnet-library-newtonsoft.json.yml:4]: or should we add new namespaces? `linking/library`?

File: nursery/use-dotnet-library-sharpvectors.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+rule:
+  meta:
+    name: use .NET library SharpVectors
+    namespace: nursery
```
Comment (mr-tz) [inline nursery/use-dotnet-library-sharpvectors.yml:4]:
```suggestion
    namespace: data-manipulation/svg
```

Comment (mr-tz) [reply nursery/use-dotnet-library-sharpvectors.yml:4]: looking for other input on this

File: nursery/use-dotnet-library-newtonsoft.json.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mr-tz) [reply nursery/use-dotnet-library-newtonsoft.json.yml:4]: or `data-manipulation/json`?

**Review by mr-tz — CHANGES_REQUESTED** [review]
This is a great start. A few suggestions and comments for discussion.

File: anti-analysis/obfuscation/use-dotnet-library-encryptdecryptutils.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    mbc:
+      - Anti-Static Analysis::Executable Code Obfuscation [B0032]
+    references:
+      - https://www.welivesecurity.com/2021/04/06/janeleiro-time-traveler-new-old-banking-trojan-brazil/
+    examples:
+      - 7f4ba9fc95b30baf8922a6933a4ff1c6a7fef41fae487bb31014c4963357770f
+  features:
+    - and:
+      - match: compiled to the .NET platform
+      - string: "RijndaelEncryptDecrypt"
```
[the remark this answers is not in the document]
Comment (johnk3r) [reply anti-analysis/obfuscation/use-dotnet-library-encryptdecryptutils.yml]: Done. About detection by class or namespace, for the samples I have this was not possible. The libraries are in the "references" section.

[the remark this answers is not in the document]
Comment (johnk3r) [reply anti-analysis/obfuscation/use-dotnet-library-encryptdecryptutils.yml]: 2022-08-25 00_04_55-use  NET libraries · mandiant_capa-rules@cd7d705

Can I disregard this?

File: nursery/use-dotnet-library-newtonsoft.json.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnk3r) [reply nursery/use-dotnet-library-newtonsoft.json.yml]: "data-manipulation/json" seems more suitable to me.

File: anti-analysis/obfuscation/use-dotnet-library-encryptdecryptutils.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (mr-tz) [reply anti-analysis/obfuscation/use-dotnet-library-encryptdecryptutils.yml]: Please quote the string again per the linter.

[the remark this answers is not in the document]
Comment (mr-tz) [reply anti-analysis/obfuscation/use-dotnet-library-encryptdecryptutils.yml]: @mike-hunhoff is this already tracked as a possible improvement? maybe not worth adding if we can just get away with string, of course.

File: collection/use-dotnet-library-sharpclipboard.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    mbc:
+      - Collection::Input Capture [E1056]
+    references:
+      - https://www.welivesecurity.com/2021/04/06/janeleiro-time-traveler-new-old-banking-trojan-brazil/
+    examples:
+      - 387f15043f0198fd3a637b0758c2b6dde9ead795c3ed70803426fc355731b173
+  features:
+    - and:
+      - match: compiled to the .NET platform
+      - string: SharpClipboard
```
Comment (mr-tz) [inline collection/use-dotnet-library-sharpclipboard.yml:19]:
```suggestion
      - string: "SharpClipboard"
```

Comment (mr-tz) [reply collection/use-dotnet-library-sharpclipboard.yml:19]: sorry, my bad!

File: data-manipulation/json/use-dotnet-library-newtonsoftjson.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+      - johnk3r
+    scope: file
+    references:
+      - https://www.welivesecurity.com/2021/04/06/janeleiro-time-traveler-new-old-banking-trojan-brazil/
+    examples:
+      - 387f15043f0198fd3a637b0758c2b6dde9ead795c3ed70803426fc355731b173
+  features:
+    - and:
+      - match: compiled to the .NET platform
+      - string: Newtonsoft.Json
```
Comment (mr-tz) [inline data-manipulation/json/use-dotnet-library-newtonsoftjson.yml:15]:
```suggestion
      - string: "Newtonsoft.Json"
```

File: data-manipulation/svg/use-dotnet-library-sharpvectors.yml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+      - johnk3r
+    scope: file
+    references:
+      - https://www.welivesecurity.com/2021/04/06/janeleiro-time-traveler-new-old-banking-trojan-brazil/
+    examples:
+      - 387f15043f0198fd3a637b0758c2b6dde9ead795c3ed70803426fc355731b173
+  features:
+    - and:
+      - match: compiled to the .NET platform
+      - string: SharpVectors.Converters.Wpf
```
Comment (mr-tz) [inline data-manipulation/svg/use-dotnet-library-sharpvectors.yml:15]:
```suggestion
      - string: "SharpVectors.Converters.Wpf"
```

File: data-manipulation/encryption/aes/use-dotnet-library-encryptdecryptutils.yml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    mbc:
+      - Anti-Static Analysis::Executable Code Obfuscation [B0032]
+    references:
+      - https://www.welivesecurity.com/2021/04/06/janeleiro-time-traveler-new-old-banking-trojan-brazil/
+    examples:
+      - 7f4ba9fc95b30baf8922a6933a4ff1c6a7fef41fae487bb31014c4963357770f
+  features:
+    - and:
+      - match: compiled to the .NET platform
+      - string: RijndaelEncryptDecrypt
```
Comment (mr-tz) [inline data-manipulation/encryption/aes/use-dotnet-library-encryptdecryptutils.yml:19]:
```suggestion
      - string: "RijndaelEncryptDecrypt"
```

File: anti-analysis/obfuscation/use-dotnet-library-encryptdecryptutils.yml [hunk 8]
```diff
@@ -0,0 +1,19 @@
+rule:
+  meta:
+    name: use .NET library EncryptDecryptUtils
+    namespace: anti-analysis/obfuscation
```
[the remark this answers is not in the document]
Comment (mr-tz) [reply anti-analysis/obfuscation/use-dotnet-library-encryptdecryptutils.yml]: Do you think this is obfuscation vs. encryption?

File: nursery/use-dotnet-library-sharpvectors.yml (hunk 2 above)
Comment (mr-tz) [reply nursery/use-dotnet-library-sharpvectors.yml]: still TODO, `nursery` is not a "valid" namespace
@mike-hunhoff?

**Review by mr-tz — CHANGES_REQUESTED** [review]
thanks, we're getting close now!

File: anti-analysis/obfuscation/use-dotnet-library-encryptdecryptutils.yml (hunk 8 above)
[the remark this answers is not in the document]
Comment (johnk3r) [reply anti-analysis/obfuscation/use-dotnet-library-encryptdecryptutils.yml]: My bad.

File: executable/resource/embed-dependencies-as-resources-using-fodycostura.yml [hunk 9]
```diff
@@ -0,0 +1,19 @@
+rule:
+  meta:
+    name: embed dependencies as resources using Fody/Costura
+    namespace: executable/resource
+    authors:
+      - johnk3r, moritz.raabe@mandiant.com
```
Comment (mr-tz) [inline executable/resource/embed-dependencies-as-resources-using-fodycostura.yml:6]:
```suggestion
      - johnk3r
      - "@mr-tz"
```

Comment (mr-tz) [reply executable/resource/embed-dependencies-as-resources-using-fodycostura.yml:6]: thanks :)

**Review by mr-tz — CHANGES_REQUESTED** [review]
one last change to make authors a list, then GTG!

File: executable/resource/embed-dependencies-as-resources-using-fodycostura.yml (hunk 9 above)
Comment (johnk3r) [reply executable/resource/embed-dependencies-as-resources-using-fodycostura.yml]: I think it's ok now!

**Review by mr-tz — APPROVED** [review]

**Comment by mr-tz** [comment]
Awesome, thanks a lot, @johnk3r!!!

**Pull request merged**